### PR TITLE
[SPARK-36865][PYTHON][DOCS] Add PySpark API document of session_window

### PR DESCRIPTION
**Pull request by sarutak**
### What changes were proposed in this pull request?

This PR adds PySpark API document of `session_window`.
The docstring of the function doesn't comply with numpydoc format so this PR also fix it.
Further, the API document of `window` doesn't have `Parameters` section so it's also added in this PR.

### Why are the changes needed?

To provide PySpark users with the API document of the newly added function.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

`make html` in `python/docs` and get the following docs.

[window]
time-window-python-doc-after

[session_window]
session-window-python-doc-after

### CODE REVIEW

File: python/pyspark/sql/functions.py [hunk 1]
```diff
@@ -2378,6 +2378,29 @@ def window(timeColumn, windowDuration, slideDuration=None, startTime=None):
 
     .. versionadded:: 2.0.0
 
+    Parameters
```
Comment (HyukjinKwon) [inline python/pyspark/sql/functions.py:2381]: Nice!

**Review by HyukjinKwon — APPROVED** [review]
Looks good. cc @HeartSaVioR FYI

**Review by HeartSaVioR — APPROVED** [review]
+1 Thanks for the work!

**Comment by HeartSaVioR** [comment]
Thanks! Merging to master/3.2.

**Pull request closed without merging**